---
id: as-011
title: "AS-011: Compatibility Too Long - Agent Skills"
sidebar_label: "AS-011"
description: "agnix rule AS-011 checks for compatibility too long in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-011", "compatibility too long", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-011`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: migrate-db
description: Use when running database migrations
compatibility: ""
---
Run the migration scripts.
```

### Valid

```markdown
---
name: migrate-db
description: Use when running database migrations
compatibility: Requires PostgreSQL 14+ and Node.js 18+
---
Run the migration scripts.
```
