---
id: cdx-000
title: "CDX-000: TOML Parse Error - Codex CLI"
sidebar_label: "CDX-000"
description: "agnix rule CDX-000 checks for toml parse error in codex cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["CDX-000", "toml parse error", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-000`
- **Severity**: `HIGH`
- **Category**: `Codex CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-09`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/openai/codex

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```toml
[model
approvalMode = suggest
```

### Valid

```toml
[model]
approvalMode = "suggest"
fullAutoErrorMode = "ask-user"
```
