---
id: agm-006
title: "AGM-006: Nested AGENTS.md Hierarchy - AGENTS.md"
sidebar_label: "AGM-006"
description: "agnix rule AGM-006 checks for nested agents.md hierarchy in agents.md files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AGM-006", "nested agents.md hierarchy", "agents.md", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AGM-006`
- **Severity**: `MEDIUM`
- **Category**: `AGENTS.md`
- **Normative Level**: `BEST_PRACTICE`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cline.bot/features/custom-instructions
- https://github.com/github/docs/changelog/2025-06-17-github-copilot-coding-agent-now-supports-agents-md-custom-instructions

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Root AGENTS.md

Project instructions.

# src/AGENTS.md (nested, separate file)

Component-specific instructions that create hierarchy ambiguity.
```

### Valid

```markdown
# Project

Single AGENTS.md at the project root with all instructions.

## Commands

- npm test
```
