---
id: agm-004
title: "AGM-004: Missing Project Context - AGENTS.md"
sidebar_label: "AGM-004"
description: "agnix rule AGM-004 checks for missing project context in agents.md files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AGM-004", "missing project context", "agents.md", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AGM-004`
- **Severity**: `MEDIUM`
- **Category**: `AGENTS.md`
- **Normative Level**: `BEST_PRACTICE`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Build Commands

Run npm install and npm build.

## Testing

Use npm test.
```

### Valid

```markdown
# Project

This is a web application for task management built with React and Node.js.

## Build Commands

Run npm install and npm build.
```
