---
id: agm-004
title: "AGM-004: Missing Project Context - AGENTS.md"
sidebar_label: "AGM-004"
description: "agnix rule AGM-004 checks for missing project context in agents.md files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AGM-004", "missing project context", "agents.md", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AGM-004`
- **Severity**: `MEDIUM`
- **Category**: `AGENTS.md`
- **Normative Level**: `BEST_PRACTICE`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Instructions
Do everything automatically.
```

### Valid

```markdown
## Project Instructions
- Use AGENTS.md as instruction entrypoint
- Keep commands explicit and test changes
```
