---
id: agm-002
title: "AGM-002: Missing Section Headers - AGENTS.md"
sidebar_label: "AGM-002"
description: "agnix rule AGM-002 checks for missing section headers in agents.md files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AGM-002", "missing section headers", "agents.md", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AGM-002`
- **Severity**: `MEDIUM`
- **Category**: `AGENTS.md`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context
- https://docs.cline.bot/features/custom-instructions

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
Just plain text without any headers.
No structure at all.
Hard to navigate.
```

### Valid

```markdown
# Main Title

Some content here.

## Section

More content.
```
