---
id: amp-003
title: "AMP-003: Invalid AGENTS.md globs Frontmatter for Amp"
sidebar_label: "AMP-003"
description: "agnix rule AMP-003 checks for invalid agents.md globs frontmatter for amp in amp checks files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AMP-003", "invalid agents.md globs frontmatter for amp", "amp checks", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AMP-003`
- **Severity**: `MEDIUM`
- **Category**: `Amp Checks`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `amp`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://ampcode.com/manual#settings

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
globs: "[broken"
---
# AGENTS

Invalid glob syntax.
```

### Valid

```markdown
---
globs:
  - "src/**/*.rs"
---
# AGENTS

Project instructions for Amp.
```
