---
id: amp-001
title: "AMP-001: Invalid Amp Check Frontmatter - Amp Checks"
sidebar_label: "AMP-001"
description: "agnix rule AMP-001 checks for invalid amp check frontmatter in amp checks files. Severity: HIGH. See examples and fix guidance."
keywords: ["AMP-001", "invalid amp check frontmatter", "amp checks", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AMP-001`
- **Severity**: `HIGH`
- **Category**: `Amp Checks`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `amp`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://ampcode.com/manual#code-review-checks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Review security-sensitive changes
unknownKey: true
---
# Security Review

Missing required name field.
```

### Valid

```markdown
---
name: security-review
description: Review security-sensitive changes
severity-default: high
tools:
  - rg
---
# Security Review

Review auth, secrets, and shell execution paths.
```
