---
id: amp-002
title: "AMP-002: Invalid Amp severity-default - Amp Checks"
sidebar_label: "AMP-002"
description: "agnix rule AMP-002 checks for invalid amp severity-default in amp checks files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AMP-002", "invalid amp severity-default", "amp checks", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AMP-002`
- **Severity**: `MEDIUM`
- **Category**: `Amp Checks`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `amp`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://ampcode.com/manual#code-review-checks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: security-review
severity-default: urgent
---
# Security Review

Invalid severity-default value.
```

### Valid

```markdown
---
name: security-review
severity-default: high
---
# Security Review

Review auth and secret handling.
```
