---
id: amp-004
title: "AMP-004: Invalid Amp Settings Configuration - Amp Checks"
sidebar_label: "AMP-004"
description: "agnix rule AMP-004 checks for invalid amp settings configuration in amp checks files. Severity: HIGH. See examples and fix guidance."
keywords: ["AMP-004", "invalid amp settings configuration", "amp checks", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AMP-004`
- **Severity**: `HIGH`
- **Category**: `Amp Checks`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `amp`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://ampcode.com/manual#settings

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "model": "amp-default",
  "unknownKey": true
}
```

### Valid

```json
{
  "model": "amp-default",
  "notify": true
}
```
