---
id: as-013
title: "AS-013: File Reference Too Deep - Agent Skills"
sidebar_label: "AS-013"
description: "agnix rule AS-013 checks for file reference too deep in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-013", "file reference too deep", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-013`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: read-config
description: Use when reading project configuration
---
See references/deep/nested/config.md for details.```

### Valid

```markdown
---
name: read-config
description: Use when reading project configuration
---
See references/config.md for details.```
