---
id: as-014
title: "AS-014: Windows Path Separator - Agent Skills"
sidebar_label: "AS-014"
description: "agnix rule AS-014 checks for windows path separator in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-014", "windows path separator", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-014`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (safe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: check-paths
description: Use when validating file paths
---
Check src\utils\helper.ts for the implementation.
```

### Valid

```markdown
---
name: check-paths
description: Use when validating file paths
---
Check src/utils/helper.ts for the implementation.
```
