---
id: as-019
title: "AS-019: Vague Skill Name - Agent Skills"
sidebar_label: "AS-019"
description: "agnix rule AS-019 checks for vague skill name in agent skills files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AS-019", "vague skill name", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-019`
- **Severity**: `MEDIUM`
- **Category**: `Agent Skills`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: helper
description: Use when handling tasks
---
Execute helper routines.
```

### Valid

```markdown
---
name: postgres-query-planner
description: Use when optimizing PostgreSQL queries
---
Analyze query plans and suggest improvements.
```
