---
id: as-006
title: "AS-006: Consecutive Hyphens in Name - Agent Skills"
sidebar_label: "AS-006"
description: "agnix rule AS-006 checks for consecutive hyphens in name in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-006", "consecutive hyphens in name", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-006`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (safe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: format--code
description: Use when formatting source code
---
Run prettier on all files.```

### Valid

```markdown
---
name: format-code
description: Use when formatting source code
---
Run prettier on all files.```
