---
id: as-018
title: "AS-018: Description Uses First or Second Person"
sidebar_label: "AS-018"
description: "agnix rule AS-018 checks for description uses first or second person in agent skills files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AS-018", "description uses first or second person", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-018`
- **Severity**: `MEDIUM`
- **Category**: `Agent Skills`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: code-review
description: You can use this skill when reviewing pull requests
---
Review code and summarize critical findings.
```

### Valid

```markdown
---
name: code-review
description: Use when reviewing pull requests for defects
---
Review code and summarize critical findings.
```
