---
id: as-017
title: "AS-017: Name Must Match Parent Directory - Agent Skills"
sidebar_label: "AS-017"
description: "agnix rule AS-017 checks for name must match parent directory in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-017", "name must match parent directory", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-017`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: deploy-helper
description: Use when validating deployment checks
---
This file is stored in code-review/SKILL.md.
```

### Valid

```markdown
---
name: code-review
description: Use when reviewing pull requests
---
Review changed files and report findings.
```
