---
id: as-001
title: "AS-001: Missing Frontmatter - Agent Skills"
sidebar_label: "AS-001"
description: "agnix rule AS-001 checks for missing frontmatter in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-001", "missing frontmatter", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-001`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
No frontmatter here.
Just plain markdown content without YAML delimiters.```

### Valid

```markdown
---
name: deploy-app
description: Use when deploying the application to production
---
Run the deploy script with the correct environment variables.```
