---
id: as-012
title: "AS-012: Content Exceeds 500 Lines - Agent Skills"
sidebar_label: "AS-012"
description: "agnix rule AS-012 checks for content exceeds 500 lines in agent skills files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AS-012", "content exceeds 500 lines", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-012`
- **Severity**: `MEDIUM`
- **Category**: `Agent Skills`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `claude-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://platform.claude.com/docs
- https://agentskills.io

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: setup-env
description: Use when setting up the development environment
---
Step 1: Install Node.js
Step 2: Install npm packages
Step 3: Configure database
Step 4: Set environment variables
Step 5: Run migrations
Step 6: Seed data
Step 7: Start server
Step 8: Verify health check
Step 9: Run smoke tests
Step 10: Done
(imagine this continues for 500+ lines)
```

### Valid

```markdown
---
name: setup-env
description: Use when setting up the development environment
---
Install dependencies and configure env vars.
```
