---
id: as-002
title: "AS-002: Missing Required Field: name - Agent Skills"
sidebar_label: "AS-002"
description: "agnix rule AS-002 checks for missing required field: name in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-002", "missing required field: name", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-002`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Use when linting the codebase
---
Run eslint on all source files.
```

### Valid

```markdown
---
name: lint-code
description: Use when linting the codebase
---
Run eslint on all source files.
```
