---
id: as-015
title: "AS-015: Upload Size Exceeds 8MB - Agent Skills"
sidebar_label: "AS-015"
description: "agnix rule AS-015 checks for upload size exceeds 8mb in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-015", "upload size exceeds 8mb", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-015`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `claude-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://platform.claude.com/docs

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: huge-skill
description: Use when running a task with many large assets
---
This skill bundles 10MB of binary data in its directory.```

### Valid

```markdown
---
name: small-skill
description: Use when running a lightweight task
---
This skill directory is under 8MB total.```
