---
id: as-003
title: "AS-003: Missing Required Field: description - Agent Skills"
sidebar_label: "AS-003"
description: "agnix rule AS-003 checks for missing required field: description in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-003", "missing required field: description", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-003`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: lint-code
---
Run eslint on all source files.```

### Valid

```markdown
---
name: lint-code
description: Use when linting the codebase
---
Run eslint on all source files.```
