---
id: cdx-000
title: "CDX-000: Codex Config Parse Error - Codex CLI"
sidebar_label: "CDX-000"
description: "agnix rule CDX-000 checks for codex config parse error in codex cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["CDX-000", "codex config parse error", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-000`
- **Severity**: `HIGH`
- **Category**: `Codex CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-09`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/openai/codex

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```text
Configuration omitted required fields for this rule.```

### Valid

```text
Configuration includes required fields and follows the rule.```
