---
id: cdx-005
title: "CDX-005: project_doc_max_bytes Exceeds Limit - Codex CLI"
sidebar_label: "CDX-005"
description: "agnix rule CDX-005 checks for project_doc_max_bytes exceeds limit in codex cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["CDX-005", "project_doc_max_bytes exceeds limit", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-005`
- **Severity**: `HIGH`
- **Category**: `Codex CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```toml
project_doc_max_bytes = 100000
```

### Valid

```toml
project_doc_max_bytes = 32768
```
