---
id: cdx-004
title: "CDX-004: Unknown Config Key - Codex CLI"
sidebar_label: "CDX-004"
description: "agnix rule CDX-004 checks for unknown config key in codex cli files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CDX-004", "unknown config key", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-004`
- **Severity**: `MEDIUM`
- **Category**: `Codex CLI`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```toml
unknownKey = true
```

### Valid

```toml
model = "o4-mini"
approvalMode = "suggest"
```
