---
id: cdx-002
title: "CDX-002: Invalid Full Auto Error Mode - Codex CLI"
sidebar_label: "CDX-002"
description: "agnix rule CDX-002 checks for invalid full auto error mode in codex cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["CDX-002", "invalid full auto error mode", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-002`
- **Severity**: `HIGH`
- **Category**: `Codex CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/openai/codex

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
fullAutoErrorMode = "crash"
```

### Valid

```text
fullAutoErrorMode = "ask-user"
```
