---
id: cln-001
title: "CLN-001: Empty Cline Rules File - Cline"
sidebar_label: "CLN-001"
description: "agnix rule CLN-001 checks for empty cline rules file in cline files. Severity: HIGH. See examples and fix guidance."
keywords: ["CLN-001", "empty cline rules file", "cline", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CLN-001`
- **Severity**: `HIGH`
- **Category**: `Cline`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `cline`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cline.bot/improving-your-workflow/cline-rules

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
# Rules
```

### Valid

```text
# Project Rules

Always follow the coding style guide and write tests for new features.
```
