---
id: cln-004
title: "CLN-004: Scalar Paths in Cline Rules - Cline"
sidebar_label: "CLN-004"
description: "agnix rule CLN-004 checks for scalar paths in cline rules in cline files. Severity: HIGH. See examples and fix guidance."
keywords: ["CLN-004", "scalar paths in cline rules", "cline", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CLN-004`
- **Severity**: `HIGH`
- **Category**: `Cline`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (safe)`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `cline`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cline.bot/features/cline-rules

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
paths: "**/*.ts"
---
# TypeScript Rules

Use strict mode.
```

### Valid

```markdown
---
paths:
  - "**/*.ts"
---
# TypeScript Rules

Use strict mode.
```
