---
id: cln-002
title: "CLN-002: Invalid Paths Glob in Cline Rules - Cline"
sidebar_label: "CLN-002"
description: "agnix rule CLN-002 checks for invalid paths glob in cline rules in cline files. Severity: HIGH. See examples and fix guidance."
keywords: ["CLN-002", "invalid paths glob in cline rules", "cline", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CLN-002`
- **Severity**: `HIGH`
- **Category**: `Cline`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-09`

## Applicability

- **Tool**: `cline`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cline.bot/features/cline-rules/overview

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```text
Configuration omitted required fields for this rule.```

### Valid

```text
Configuration includes required fields and follows the rule.```
