---
id: cln-003
title: "CLN-003: Unknown Frontmatter Key in Cline Rules - Cline"
sidebar_label: "CLN-003"
description: "agnix rule CLN-003 checks for unknown frontmatter keys in cline rules files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CLN-003", "unknown frontmatter key in cline rules", "cline", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CLN-003`
- **Severity**: `MEDIUM`
- **Category**: `Cline`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `cline`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cline.bot/features/cline-rules

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
---
paths: "**/*.ts"
unknownKey: value
---
# TypeScript Rules
```

### Valid

```markdown
---
paths: "**/*.ts"
---
# TypeScript Rules
```
