---
id: cop-010
title: "COP-010: Custom Agent Uses Deprecated infer Field"
sidebar_label: "COP-010"
description: "agnix rule COP-010 checks for custom agent uses deprecated infer field in github copilot files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["COP-010", "custom agent uses deprecated infer field", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-010`
- **Severity**: `MEDIUM`
- **Category**: `GitHub Copilot`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/avifenesh/agnix/issues/400

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Review pull requests
infer: true
---
Review pull requests.
```

### Valid

```markdown
---
description: Review pull requests
---
Review pull requests.
```
