---
id: cop-018
title: "COP-018: Copilot Setup Steps Missing or Invalid copilot-setup-steps Job"
sidebar_label: "COP-018"
description: "agnix rule COP-018 checks for copilot setup steps missing or invalid copilot-setup-steps job in github copilot files. Severity: HIGH. See examples and fix gu..."
keywords: ["COP-018", "copilot setup steps missing or invalid copilot-setup-steps job", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-018`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/how-tos/agents/copilot-coding-agent/customizing-the-development-environment-for-copilot-coding-agent

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
jobs:
  setup:
    runs-on: ubuntu-latest
    steps:
      - run: echo setup
```

### Valid

```markdown
jobs:
  copilot-setup-steps:
    runs-on: ubuntu-latest
    steps:
      - run: echo setup
```
