---
id: cop-002
title: "COP-002: Invalid Frontmatter in Scoped Instructions"
sidebar_label: "COP-002"
description: "agnix rule COP-002 checks for invalid frontmatter in scoped instructions in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-002", "invalid frontmatter in scoped instructions", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-002`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/customizing-copilot

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Copilot Instructions
Write whatever code seems fine.
```

### Valid

```markdown
# Copilot Instructions
Use project coding standards and keep tests updated.
```
