---
id: cop-008
title: "COP-008: Custom Agent Unknown Frontmatter Field"
sidebar_label: "COP-008"
description: "agnix rule COP-008 checks for custom agent unknown frontmatter field in github copilot files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["COP-008", "custom agent unknown frontmatter field", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-008`
- **Severity**: `MEDIUM`
- **Category**: `GitHub Copilot`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/reference/custom-agents-configuration

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Review pull requests
mystery-field: true
---
Review pull requests.
```

### Valid

```markdown
---
description: Review pull requests
target: vscode
---
Review pull requests.
```
