---
id: cop-009
title: "COP-009: Custom Agent Invalid Target - GitHub Copilot"
sidebar_label: "COP-009"
description: "agnix rule COP-009 checks for custom agent invalid target in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-009", "custom agent invalid target", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-009`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/reference/custom-agents-configuration

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Review pull requests
target: desktop
---
Review pull requests.
```

### Valid

```markdown
---
description: Review pull requests
target: vscode
---
Review pull requests.
```
