---
id: cop-017
title: "COP-017: Copilot Hooks Schema Validation - GitHub Copilot"
sidebar_label: "COP-017"
description: "agnix rule COP-017 checks for copilot hooks schema validation in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-017", "copilot hooks schema validation", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-017`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/reference/hooks-configuration

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{"version":1,"hooks":[{"events":["sessionStart"],"command":{"bash":"echo start"}}]}
```

### Valid

```json
{"version":1,"hooks":[{"type":"command","events":["sessionStart"],"command":{"bash":"echo start"}}]}
```
