---
id: cop-012
title: "COP-012: Custom Agent Uses GitHub.com Unsupported Fields"
sidebar_label: "COP-012"
description: "agnix rule COP-012 checks for custom agent uses github.com unsupported fields in github copilot files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["COP-012", "custom agent uses github.com unsupported fields", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-012`
- **Severity**: `MEDIUM`
- **Category**: `GitHub Copilot`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/reference/custom-agents-configuration

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Review pull requests
model: gpt-4.1
argument-hint: branch-name
---
Review pull requests.
```

### Valid

```markdown
---
description: Review pull requests
---
Review pull requests.
```
