---
id: cop-015
title: "COP-015: Prompt File Invalid Agent Mode - GitHub Copilot"
sidebar_label: "COP-015"
description: "agnix rule COP-015 checks for prompt file invalid agent mode in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-015", "prompt file invalid agent mode", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-015`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://code.visualstudio.com/docs/copilot/customization/prompt-files

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Refactor selected code
agent: maybe
---
Refactor the selected code.
```

### Valid

```markdown
---
description: Refactor selected code
agent: ask
---
Refactor the selected code.
```
