---
id: cop-013
title: "COP-013: Prompt File Empty Body - GitHub Copilot"
sidebar_label: "COP-013"
description: "agnix rule COP-013 checks for prompt file empty body in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-013", "prompt file empty body", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-013`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://code.visualstudio.com/docs/copilot/customization/prompt-files

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Refactor selected code
---
```

### Valid

```markdown
---
description: Refactor selected code
---
Refactor the selected code while preserving behavior.
```
