---
id: cop-007
title: "COP-007: Custom Agent Missing Description - GitHub Copilot"
sidebar_label: "COP-007"
description: "agnix rule COP-007 checks for custom agent missing description in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-007", "custom agent missing description", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-007`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/reference/custom-agents-configuration

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
target: vscode
---
Review pull requests.
```

### Valid

```markdown
---
description: Review pull requests for quality and safety
---
Review pull requests and suggest focused improvements.
```
