---
id: cop-014
title: "COP-014: Prompt File Unknown Frontmatter Field"
sidebar_label: "COP-014"
description: "agnix rule COP-014 checks for prompt file unknown frontmatter field in github copilot files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["COP-014", "prompt file unknown frontmatter field", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-014`
- **Severity**: `MEDIUM`
- **Category**: `GitHub Copilot`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://code.visualstudio.com/docs/copilot/customization/prompt-files

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: Refactor selected code
unknown-option: true
---
Refactor the selected code.
```

### Valid

```markdown
---
description: Refactor selected code
agent: ask
---
Refactor the selected code.
```
