---
id: cop-005
title: "COP-005: Invalid excludeAgent Value - GitHub Copilot"
sidebar_label: "COP-005"
description: "agnix rule COP-005 checks for invalid excludeagent value in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-005", "invalid excludeagent value", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-005`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/customizing-copilot

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
applyTo: "**/*.ts"
excludeAgent: "invalid-agent"
---
# TypeScript Instructions

Use strict mode.
```

### Valid

```markdown
---
applyTo: "**/*.ts"
excludeAgent: "code-review"
---
# TypeScript Instructions

Use strict mode.
```
