---
id: cur-005
title: "CUR-005: Unknown Frontmatter Keys - Cursor"
sidebar_label: "CUR-005"
description: "agnix rule CUR-005 checks for unknown frontmatter keys in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-005", "unknown frontmatter keys", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-005`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Rule
No metadata block
```

### Valid

```markdown
---
description: Cursor rule for repository policy
---
Use project-specific guidance.
```
