---
id: cur-016
title: "CUR-016: Invalid Cursor Environment Schema - Cursor"
sidebar_label: "CUR-016"
description: "agnix rule CUR-016 checks for invalid cursor environment schema in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-016", "invalid cursor environment schema", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-016`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/cloud-agent

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{"snapshot":42,"install":"npm ci","terminals":[{"name":"app"}]}
```

### Valid

```markdown
{"snapshot":"ubuntu-24.04","install":"npm ci","start":"npm run dev","terminals":[{"name":"app","command":"npm run dev"}]}
```
