---
id: cur-002
title: "CUR-002: Missing Frontmatter in .mdc File - Cursor"
sidebar_label: "CUR-002"
description: "agnix rule CUR-002 checks for missing frontmatter in .mdc file in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-002", "missing frontmatter in .mdc file", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-002`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# TypeScript Rules

Use strict mode and explicit types.```

### Valid

```markdown
---
description: TypeScript rules
globs: "**/*.ts"
---
# TypeScript Rules

Use strict mode.```
