---
id: cur-006
title: "CUR-006: Legacy .cursorrules File Detected - Cursor"
sidebar_label: "CUR-006"
description: "agnix rule CUR-006 checks for legacy .cursorrules file detected in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-006", "legacy .cursorrules file detected", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-006`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Rule
No metadata block
```

### Valid

```markdown
---
description: Cursor rule for repository policy
---
Use project-specific guidance.
```
