---
id: cur-015
title: "CUR-015: Empty Cursor Subagent Body - Cursor"
sidebar_label: "CUR-015"
description: "agnix rule CUR-015 checks for empty cursor subagent body in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-015", "empty cursor subagent body", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-015`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/context/subagents

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: reviewer-agent
description: Reviews code for defects
---
```

### Valid

```markdown
---
name: reviewer-agent
description: Reviews code for defects
---
Review the diff and return a concise risk report.
```
