---
id: cur-005
title: "CUR-005: Unknown Frontmatter Keys - Cursor"
sidebar_label: "CUR-005"
description: "agnix rule CUR-005 checks for unknown frontmatter keys in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-005", "unknown frontmatter keys", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-005`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (safe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: TypeScript rules
unknownKey: some-value
---
# Rules

Use strict mode.
```

### Valid

```markdown
---
description: TypeScript rules
globs: "**/*.ts"
alwaysApply: false
---
# Rules

Use strict mode.
```
