---
id: cur-008
title: "CUR-008: Invalid alwaysApply Type - Cursor"
sidebar_label: "CUR-008"
description: "agnix rule CUR-008 checks for invalid alwaysapply type in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-008", "invalid alwaysapply type", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-008`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (safe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: TypeScript rules
alwaysApply: "true"
---
# Rules

Always use strict mode.```

### Valid

```markdown
---
description: TypeScript rules
alwaysApply: true
---
# Rules

Always use strict mode.```
