---
id: cur-001
title: "CUR-001: Empty Cursor Rule File - Cursor"
sidebar_label: "CUR-001"
description: "agnix rule CUR-001 checks for empty cursor rule file in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-001", "empty cursor rule file", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-001`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Rule
No metadata block
```

### Valid

```markdown
---
description: Cursor rule for repository policy
---
Use project-specific guidance.
```
