---
id: cur-012
title: "CUR-012: Hook Entry Missing Required Command Field - Cursor"
sidebar_label: "CUR-012"
description: "agnix rule CUR-012 checks for hook entry missing required command field in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-012", "hook entry missing required command field", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-012`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/agent/hooks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{"version":1,"hooks":{"sessionStart":[{"type":"command"}]}}
```

### Valid

```markdown
{"version":1,"hooks":{"sessionStart":[{"type":"command","command":"echo start"}]}}
```
