---
id: cur-013
title: "CUR-013: Invalid Cursor Hook Type Value - Cursor"
sidebar_label: "CUR-013"
description: "agnix rule CUR-013 checks for invalid cursor hook type value in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-013", "invalid cursor hook type value", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-013`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/agent/hooks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{"version":1,"hooks":{"sessionStart":[{"type":"agent","command":"echo start"}]}}
```

### Valid

```json
{"version":1,"hooks":{"sessionStart":[{"type":"prompt","command":"Summarize changes"}]}}
```
