---
id: cur-009
title: "CUR-009: Missing Description for Agent-Requested Rule"
sidebar_label: "CUR-009"
description: "agnix rule CUR-009 checks for missing description for agent-requested rule in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-009", "missing description for agent-requested rule", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-009`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Rule
No metadata block
```

### Valid

```markdown
---
description: Cursor rule for repository policy
---
Use project-specific guidance.
```
