---
id: cur-010
title: "CUR-010: Invalid Cursor Hooks Schema - Cursor"
sidebar_label: "CUR-010"
description: "agnix rule CUR-010 checks for invalid cursor hooks schema in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-010", "invalid cursor hooks schema", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-010`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/agent/hooks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{"hooks":{}}
```

### Valid

```markdown
{"version":1,"hooks":{"sessionStart":[{"type":"command","command":"echo start"}]}}
```
