---
id: cur-014
title: "CUR-014: Invalid Cursor Subagent Frontmatter - Cursor"
sidebar_label: "CUR-014"
description: "agnix rule CUR-014 checks for invalid cursor subagent frontmatter in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-014", "invalid cursor subagent frontmatter", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-014`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/context/subagents

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: ReviewerAgent
description: 123
readonly: "true"
---
Review staged changes.
```

### Valid

```markdown
---
name: reviewer-agent
description: Reviews code for defects
model: fast
readonly: true
is_background: false
---
Review staged changes and prioritize findings.
```
