---
id: cur-011
title: "CUR-011: Unknown Cursor Hook Event Name - Cursor"
sidebar_label: "CUR-011"
description: "agnix rule CUR-011 checks for unknown cursor hook event name in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-011", "unknown cursor hook event name", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-011`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://cursor.com/docs/agent/hooks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{"version":1,"hooks":{"unknownEvent":[{"type":"command","command":"echo start"}]}}
```

### Valid

```markdown
{"version":1,"hooks":{"sessionStart":[{"type":"command","command":"echo start"}]}}
```
