---
id: cur-004
title: "CUR-004: Invalid Glob Pattern in globs Field - Cursor"
sidebar_label: "CUR-004"
description: "agnix rule CUR-004 checks for invalid glob pattern in globs field in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-004", "invalid glob pattern in globs field", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-004`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
description: TypeScript rules
globs: "[unclosed"
---
# Rules

Use strict mode.```

### Valid

```markdown
---
description: TypeScript rules
globs: "**/*.ts"
---
# Rules

Use strict mode.```
