---
id: gm-002
title: "GM-002: Missing Section Headers in GEMINI.md - gemini-cli"
sidebar_label: "GM-002"
description: "agnix rule GM-002 checks for missing section headers in gemini-cli files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["GM-002", "missing section headers in gemini.md", "gemini-cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-002`
- **Severity**: `MEDIUM`
- **Category**: `gemini-cli`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/cli/gemini-md/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
Just plain text without any headers or structure.
Use TypeScript and write tests.
```

### Valid

```markdown
# Project Instructions

## Build Commands

Run npm install and npm build.
```
