---
id: gm-001
title: "GM-001: Invalid Markdown Structure in GEMINI.md - gemini-cli"
sidebar_label: "GM-001"
description: "agnix rule GM-001 checks for invalid markdown structure in gemini.md in gemini-cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["GM-001", "invalid markdown structure in gemini.md", "gemini-cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-001`
- **Severity**: `HIGH`
- **Category**: `gemini-cli`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/cli/gemini-md/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```text
Configuration omitted required fields for this rule.```

### Valid

```text
Configuration includes required fields and follows the rule.```
