---
id: gm-004
title: "GM-004: Invalid Hooks Configuration in Gemini Settings"
sidebar_label: "GM-004"
description: "agnix rule GM-004 checks for invalid hooks configuration in gemini settings in gemini cli files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["GM-004", "invalid hooks configuration in gemini settings", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-004`
- **Severity**: `MEDIUM`
- **Category**: `Gemini CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/hooks

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{"hooksConfig":{"InvalidEvent":[{"type":"command","command":"echo test"}]}}
```

### Valid

```json
{"hooksConfig":{"BeforeAgent":[{"type":"command","command":"echo test"}]}}
```
