---
id: gm-009
title: "GM-009: Settings.json Parse Error - Gemini CLI"
sidebar_label: "GM-009"
description: "agnix rule GM-009 checks for settings.json parse error in gemini cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["GM-009", "settings.json parse error", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-009`
- **Severity**: `HIGH`
- **Category**: `Gemini CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/cli/settings

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{ invalid json }
```

### Valid

```markdown
{"general":{},"model":{}}
```
