---
id: gm-008
title: "GM-008: Invalid Context File Name Configuration - Gemini CLI"
sidebar_label: "GM-008"
description: "agnix rule GM-008 checks for invalid context file name configuration in gemini cli files. Severity: LOW. See examples and fix guidance."
keywords: ["GM-008", "invalid context file name configuration", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-008`
- **Severity**: `LOW`
- **Category**: `Gemini CLI`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/extensions/reference

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{"contextFileName":"docs/context.md"}
```

### Valid

```markdown
{"contextFileName":"CONTEXT.md"}
```
