---
id: gm-006
title: "GM-006: Invalid .geminiignore File - Gemini CLI"
sidebar_label: "GM-006"
description: "agnix rule GM-006 checks for invalid .geminiignore file in gemini cli files. Severity: LOW. See examples and fix guidance."
keywords: ["GM-006", "invalid .geminiignore file", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-006`
- **Severity**: `LOW`
- **Category**: `Gemini CLI`
- **Normative Level**: `MAY`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/cli/settings

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Gemini
```

### Valid

```markdown
node_modules/
*.log
.env
```
