---
id: gm-003
title: "GM-003: Missing Project Context in GEMINI.md - Gemini CLI"
sidebar_label: "GM-003"
description: "agnix rule GM-003 checks for missing project context in gemini.md in gemini cli files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["GM-003", "missing project context in gemini.md", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-003`
- **Severity**: `MEDIUM`
- **Category**: `Gemini CLI`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/cli/gemini-md/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
# Build Commands

Run npm install and npm build.

## Testing

Use npm test.
```

### Valid

```text
# Project

This is a linter for agent configurations.

## Build

Run cargo build.
```
