---
id: gm-005
title: "GM-005: Invalid Extension Manifest - Gemini CLI"
sidebar_label: "GM-005"
description: "agnix rule GM-005 checks for invalid extension manifest in gemini cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["GM-005", "invalid extension manifest", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-005`
- **Severity**: `HIGH`
- **Category**: `Gemini CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/extensions/reference

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
{"version":"1.0.0"}
```

### Valid

```markdown
{"name":"my-extension","version":"1.0.0","description":"A useful extension"}
```
