---
id: gm-007
title: "GM-007: @import File Not Found in GEMINI.md - Gemini CLI"
sidebar_label: "GM-007"
description: "agnix rule GM-007 checks for @import file not found in gemini.md in gemini cli files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["GM-007", "@import file not found in gemini.md", "gemini cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `GM-007`
- **Severity**: `MEDIUM`
- **Category**: `Gemini CLI`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `gemini-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://geminicli.com/docs/cli/gemini-md/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
@import nonexistent-file.md
```

### Valid

```markdown
@import shared-rules.md
```
