---
id: mcp-015
title: "MCP-015: Missing Resource Required Fields - MCP"
sidebar_label: "MCP-015"
description: "agnix rule MCP-015 checks for missing resource required fields in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-015", "missing resource required fields", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-015`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/server/resources

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "resources": [{
    "description": "missing uri and name"
  }]
}
```

### Valid

```json
{
  "resources": [{
    "uri": "file:///README.md",
    "name": "readme"
  }]
}
```
