---
id: mcp-016
title: "MCP-016: Missing Prompt Required Name - MCP"
sidebar_label: "MCP-016"
description: "agnix rule MCP-016 checks for missing prompt required name in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-016", "missing prompt required name", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-016`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/server/prompts

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "prompts": [{
    "description": "missing prompt name"
  }]
}
```

### Valid

```json
{
  "prompts": [{
    "name": "review_code",
    "description": "Review the provided code"
  }]
}
```
