---
id: mcp-007
title: "MCP-007: MCP Parse Error - MCP"
sidebar_label: "MCP-007"
description: "agnix rule MCP-007 checks for mcp parse error in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-007", "mcp parse error", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-007`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{ invalid json content
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "method": "tools/list",
  "id": 1
}
```
