---
id: mcp-009
title: "MCP-009: Missing command for stdio server - MCP"
sidebar_label: "MCP-009"
description: "agnix rule MCP-009 checks for missing command for stdio server in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-009", "missing command for stdio server", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-009`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "my-server": {
      "type": "stdio",
      "args": ["server.js"]
    }
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "my-server": {
      "type": "stdio",
      "command": "node",
      "args": ["server.js"]
    }
  }
}
```
