---
id: mcp-013
title: "MCP-013: Invalid Tool Name Format - MCP"
sidebar_label: "MCP-013"
description: "agnix rule MCP-013 checks for invalid tool name format in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-013", "invalid tool name format", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-013`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/server/tools

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "tools": [{
    "name": "read file",
    "description": "Reads file contents",
    "inputSchema": { "type": "object" }
  }]
}
```

### Valid

```json
{
  "tools": [{
    "name": "read_file.v2",
    "description": "Reads file contents",
    "inputSchema": { "type": "object" }
  }]
}
```
