---
id: mcp-005
title: "MCP-005: Tool Without User Consent - MCP"
sidebar_label: "MCP-005"
description: "agnix rule MCP-005 checks for tool without user consent in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-005", "tool without user consent", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-005`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-06-18`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```json
{
  "jsonrpc": "1.0",
  "tools": []
}
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "tools": [
    {
      "name": "validate_file",
      "description": "Validate one configuration file",
      "inputSchema": {"type": "object"}
    }
  ]
}
```
