---
id: mcp-008
title: "MCP-008: Protocol Version Mismatch - MCP"
sidebar_label: "MCP-008"
description: "agnix rule MCP-008 checks for protocol version mismatch in mcp files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["MCP-008", "protocol version mismatch", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-008`
- **Severity**: `MEDIUM`
- **Category**: `MCP`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "jsonrpc": "2.0",
  "method": "initialize",
  "id": 1,
  "params": {
    "protocolVersion": "2024-11-05",
    "clientInfo": { "name": "my-client", "version": "1.0.0" }
  }
}
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "method": "initialize",
  "id": 1,
  "params": {
    "protocolVersion": "2025-11-25",
    "clientInfo": { "name": "my-client", "version": "1.0.0" }
  }
}
```
