---
id: mcp-012
title: "MCP-012: Deprecated SSE transport - MCP"
sidebar_label: "MCP-012"
description: "agnix rule MCP-012 checks for deprecated sse transport in mcp files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["MCP-012", "deprecated sse transport", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-012`
- **Severity**: `MEDIUM`
- **Category**: `MCP`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-03-26`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "remote-api": {
      "type": "sse",
      "url": "http://localhost:3000/sse"
    }
  }
}```

### Valid

```json
{
  "mcpServers": {
    "remote-api": {
      "type": "http",
      "url": "http://localhost:3000/mcp"
    }
  }
}```
