---
id: mcp-018
title: "MCP-018: Potential Plaintext Secret in MCP Env - MCP"
sidebar_label: "MCP-018"
description: "agnix rule MCP-018 checks for potential plaintext secret in mcp env in mcp files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["MCP-018", "potential plaintext secret in mcp env", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-018`
- **Severity**: `MEDIUM`
- **Category**: `MCP`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/basic/security_best_practices

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "local": {
      "type": "stdio",
      "command": "node",
      "env": {
        "API_KEY": "hardcoded-secret"
      }
    }
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "local": {
      "type": "stdio",
      "command": "node",
      "env": {
        "API_KEY": "${API_KEY}"
      }
    }
  }
}
```
