---
id: mcp-024
title: "MCP-024: Empty MCP Server Configuration - MCP"
sidebar_label: "MCP-024"
description: "agnix rule MCP-024 checks for empty mcp server configuration in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-024", "empty mcp server configuration", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-024`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/basic/transports

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "empty": {}
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "local": {
      "type": "stdio",
      "command": "node"
    }
  }
}
```
