---
id: mcp-023
title: "MCP-023: Duplicate MCP Server Names - MCP"
sidebar_label: "MCP-023"
description: "agnix rule MCP-023 checks for duplicate mcp server names in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-023", "duplicate mcp server names", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-023`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/basic/transports

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "dup": {"type": "stdio", "command": "node"},
    "dup": {"type": "stdio", "command": "python"}
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "local": {"type": "stdio", "command": "node"},
    "remote": {"type": "http", "url": "https://api.example.com/mcp"}
  }
}
```
