---
id: mcp-021
title: "MCP-021: Wildcard HTTP Interface Binding - MCP"
sidebar_label: "MCP-021"
description: "agnix rule MCP-021 checks for wildcard http interface binding in mcp files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["MCP-021", "wildcard http interface binding", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-021`
- **Severity**: `MEDIUM`
- **Category**: `MCP`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/basic/security_best_practices

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "wild": {
      "type": "http",
      "url": "http://0.0.0.0:3000/mcp"
    }
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "local": {
      "type": "http",
      "url": "http://localhost:3000/mcp"
    }
  }
}
```
