---
id: mcp-005
title: "MCP-005: Tool Without User Consent - MCP"
sidebar_label: "MCP-005"
description: "agnix rule MCP-005 checks for tool without user consent in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-005", "tool without user consent", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-005`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-06-18`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "name": "delete-file",
  "description": "Permanently deletes a file from the filesystem",
  "inputSchema": { "type": "object" }
}
```

### Valid

```json
{
  "name": "delete-file",
  "description": "Permanently deletes a file from the filesystem",
  "inputSchema": { "type": "object" },
  "requiresApproval": true
}
```
