---
id: mcp-020
title: "MCP-020: Unknown Capability Declaration Key - MCP"
sidebar_label: "MCP-020"
description: "agnix rule MCP-020 checks for unknown capability declaration key in mcp files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["MCP-020", "unknown capability declaration key", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-020`
- **Severity**: `MEDIUM`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/basic/lifecycle

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "capabilities": {
    "tools": {"listChanged": true},
    "unknownCapability": {}
  }
}
```

### Valid

```json
{
  "capabilities": {
    "tools": {"listChanged": true},
    "resources": {"subscribe": false}
  }
}
```
