---
id: mcp-003
title: "MCP-003: Invalid JSON Schema - MCP"
sidebar_label: "MCP-003"
description: "agnix rule MCP-003 checks for invalid json schema in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-003", "invalid json schema", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-003`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "name": "search",
  "description": "Searches for files matching a query string",
  "inputSchema": {
    "type": "invalid_type"
  },
  "outputSchema": {
    "type": 42
  }
}
```

### Valid

```json
{
  "name": "search",
  "description": "Searches for files matching a query string",
  "inputSchema": {
    "type": "object",
    "properties": {
      "query": { "type": "string" }
    }
  },
  "outputSchema": {
    "$schema": "https://json-schema.org/draft/2020-12/schema",
    "type": "object"
  }
}
```
