---
id: mcp-012
title: "MCP-012: Deprecated SSE transport - MCP"
sidebar_label: "MCP-012"
description: "agnix rule MCP-012 checks for deprecated sse transport in mcp files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["MCP-012", "deprecated sse transport", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-012`
- **Severity**: `MEDIUM`
- **Category**: `MCP`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-03-26`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```json
{
  "jsonrpc": "1.0",
  "tools": []
}
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "tools": [
    {
      "name": "validate_file",
      "description": "Validate one configuration file",
      "inputSchema": {"type": "object"}
    }
  ]
}
```
