---
id: mcp-014
title: "MCP-014: Invalid outputSchema Definition - MCP"
sidebar_label: "MCP-014"
description: "agnix rule MCP-014 checks for invalid outputschema definition in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-014", "invalid outputschema definition", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-014`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/server/tools

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "tools": [{
    "name": "lookup",
    "description": "Looks up records",
    "inputSchema": { "type": "object" },
    "outputSchema": { "type": "invalid_type" }
  }]
}
```

### Valid

```json
{
  "tools": [{
    "name": "lookup",
    "description": "Looks up records",
    "inputSchema": { "type": "object" },
    "outputSchema": { "type": "object" }
  }]
}
```
