---
id: mcp-006
title: "MCP-006: Untrusted Annotations - MCP"
sidebar_label: "MCP-006"
description: "agnix rule MCP-006 checks for untrusted annotations in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-006", "untrusted annotations", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-006`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-06-18`

## Evidence Sources

- https://modelcontextprotocol.io/docs/concepts/tools

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "name": "read-file",
  "description": "Reads a file and returns its contents as text",
  "inputSchema": { "type": "object" },
  "requiresApproval": true,
  "annotations": { "readOnlyHint": true, "title": "File Reader" }
}
```

### Valid

```json
{
  "name": "read-file",
  "description": "Reads a file and returns its contents as text",
  "inputSchema": { "type": "object" },
  "requiresApproval": true
}
```
