---
id: mcp-022
title: "MCP-022: Invalid args Array Type - MCP"
sidebar_label: "MCP-022"
description: "agnix rule MCP-022 checks for invalid args array type in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-022", "invalid args array type", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-022`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification/2025-11-25/basic/transports

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "local": {
      "type": "stdio",
      "command": "node",
      "args": "server.js --port 3000"
    }
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "local": {
      "type": "stdio",
      "command": "node",
      "args": ["server.js", "--port", "3000"]
    }
  }
}
```
