---
id: mcp-002
title: "MCP-002: Missing Required Tool Field - MCP"
sidebar_label: "MCP-002"
description: "agnix rule MCP-002 checks for missing required tool field in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-002", "missing required tool field", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-002`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-06-18`

## Evidence Sources

- https://modelcontextprotocol.io/docs/concepts/tools

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```json
{
  "jsonrpc": "1.0",
  "tools": []
}
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "tools": [
    {
      "name": "validate_file",
      "description": "Validate one configuration file",
      "inputSchema": {"type": "object"}
    }
  ]
}
```
