---
id: mcp-010
title: "MCP-010: Missing url for http/sse server - MCP"
sidebar_label: "MCP-010"
description: "agnix rule MCP-010 checks for missing url for http/sse server in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-010", "missing url for http/sse server", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-010`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-13`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-11-25`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": {
    "remote-api": {
      "type": "http"
    }
  }
}
```

### Valid

```json
{
  "mcpServers": {
    "remote-api": {
      "type": "http",
      "url": "http://localhost:3000/mcp"
    }
  }
}
```
