---
id: oc-002
title: "OC-002: Invalid Instruction Path - OpenCode"
sidebar_label: "OC-002"
description: "agnix rule OC-002 checks for invalid instruction path in opencode files. Severity: HIGH. See examples and fix guidance."
keywords: ["OC-002", "invalid instruction path", "opencode", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `OC-002`
- **Severity**: `HIGH`
- **Category**: `OpenCode`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `opencode`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://opencode.ai/docs/config

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
{
  "instructions": ["[unclosed"]
}
```

### Valid

```text
{
  "instructions": ["**/*.md"]
}
```
