---
id: oc-001
title: "OC-001: Invalid Share Mode - OpenCode"
sidebar_label: "OC-001"
description: "agnix rule OC-001 checks for invalid share mode in opencode files. Severity: HIGH. See examples and fix guidance."
keywords: ["OC-001", "invalid share mode", "opencode", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `OC-001`
- **Severity**: `HIGH`
- **Category**: `OpenCode`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `opencode`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://opencode.ai/docs/config

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
{
  "share": "public"
}
```

### Valid

```text
{
  "share": "manual"
}
```
