---
id: oc-007
title: "OC-007: Invalid Agent Definition - OpenCode"
sidebar_label: "OC-007"
description: "agnix rule OC-007 checks for invalid agent definition in opencode files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["OC-007", "invalid agent definition", "opencode", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `OC-007`
- **Severity**: `MEDIUM`
- **Category**: `OpenCode`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `opencode`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://opencode.ai/docs/config

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "agent": {
    "my-agent": {}
  }
}
```

### Valid

```json
{
  "agent": {
    "my-agent": {
      "description": "A custom agent"
    }
  }
}
```
