---
id: oc-006
title: "OC-006: Remote URL in Instructions - OpenCode"
sidebar_label: "OC-006"
description: "agnix rule OC-006 checks for remote url in instructions in opencode files. Severity: LOW. See examples and fix guidance."
keywords: ["OC-006", "remote url in instructions", "opencode", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `OC-006`
- **Severity**: `LOW`
- **Category**: `OpenCode`
- **Normative Level**: `BEST_PRACTICE`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `opencode`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://opencode.ai/docs/config

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "instructions": ["https://example.com/rules.md"]
}
```

### Valid

```json
{
  "instructions": ["**/*.md"]
}
```
