---
id: oc-009
title: "OC-009: Invalid Variable Substitution - OpenCode"
sidebar_label: "OC-009"
description: "agnix rule OC-009 checks for invalid variable substitution in opencode files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["OC-009", "invalid variable substitution", "opencode", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `OC-009`
- **Severity**: `MEDIUM`
- **Category**: `OpenCode`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `opencode`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://opencode.ai/docs/config

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "model": "{bad:value}"
}
```

### Valid

```json
{
  "model": "{env:OPENAI_MODEL}"
}
```
