---
id: pe-006
title: "PE-006: Negative-Only Instructions - Prompt Engineering"
sidebar_label: "PE-006"
description: "agnix rule PE-006 checks for negative-only instructions in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-006", "negative-only instructions", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-006`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2025-05-01`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/overview

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Rules

Don't use global variables.
Never use console.log.
Avoid inline styles.
```

### Valid

```markdown
# Rules

Don't use global variables. Instead, pass values as function parameters.
Avoid console.log in production. Use the structured logger instead.
```
