---
id: pe-002
title: "PE-002: Chain-of-Thought on Simple Task - Prompt Engineering"
sidebar_label: "PE-002"
description: "agnix rule PE-002 checks for chain-of-thought on simple task in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-002", "chain-of-thought on simple task", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-002`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://arxiv.org/abs/2201.11903

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Read File Skill

When asked to read the file, think step by step.```

### Valid

```markdown
# Code Review Skill

Think step by step when reviewing complex code.```
