---
id: pe-003
title: "PE-003: Weak Imperative Language - Prompt Engineering"
sidebar_label: "PE-003"
description: "agnix rule PE-003 checks for weak imperative language in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-003", "weak imperative language", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-003`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://arxiv.org/abs/2201.11903
- https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/overview

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Critical Rules

You should follow the style guide.
Try to run linting before commits.```

### Valid

```markdown
# Critical Rules

You must follow the style guide.
Always run linting before commits.```
