---
id: pe-003
title: "PE-003: Weak Imperative Language - Prompt Engineering"
sidebar_label: "PE-003"
description: "agnix rule PE-003 checks for weak imperative language in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-003", "weak imperative language", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-003`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://arxiv.org/abs/2201.11903
- https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/overview

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
Do the task quickly.
```

### Valid

```markdown
## Objective
Validate AGENTS.md files for schema and policy compliance.

## Output Format
Return JSON diagnostics grouped by file.
```
