---
id: pe-005
title: "PE-005: Redundant Generic Instructions - Prompt Engineering"
sidebar_label: "PE-005"
description: "agnix rule PE-005 checks for redundant generic instructions in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-005", "redundant generic instructions", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-005`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2025-05-01`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/overview

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Rules

Be helpful and accurate.
Be concise.
Format all output as JSON.
```

### Valid

```markdown
# Rules

Format all output as JSON with 2-space indentation.
Always include error codes in responses.
```
