---
id: pe-001
title: "PE-001: Lost in the Middle - Prompt Engineering"
sidebar_label: "PE-001"
description: "agnix rule PE-001 checks for lost in the middle in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-001", "lost in the middle", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-001`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://aclanthology.org/2024.tacl-1.9/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
Do the task quickly.
```

### Valid

```markdown
## Objective
Validate AGENTS.md files for schema and policy compliance.

## Output Format
Return JSON diagnostics grouped by file.
```
