---
id: ref-003
title: "REF-003: Duplicate Import - References"
sidebar_label: "REF-003"
description: "agnix rule REF-003 checks for duplicate import in references files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["REF-003", "duplicate import", "references", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `REF-003`
- **Severity**: `MEDIUM`
- **Category**: `References`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2025-05-01`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://code.claude.com/docs/en/memory

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Project

@docs/coding-standards.md
@docs/testing-guide.md
@docs/coding-standards.md
```

### Valid

```markdown
# Project

@docs/coding-standards.md
@docs/testing-guide.md
```
