---
id: ref-004
title: "REF-004: Non-Markdown Import - References"
sidebar_label: "REF-004"
description: "agnix rule REF-004 checks for non-markdown import in references files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["REF-004", "non-markdown import", "references", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `REF-004`
- **Severity**: `MEDIUM`
- **Category**: `References`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2025-05-01`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://code.claude.com/docs/en/memory

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Project

@config.json
@scripts/deploy.py
@src/utils.ts
```

### Valid

```markdown
# Project

@docs/coding-standards.md
@docs/architecture.md
```
