---
id: ref-002
title: "REF-002: Broken Markdown Link - References"
sidebar_label: "REF-002"
description: "agnix rule REF-002 checks for broken markdown link in references files. Severity: HIGH. See examples and fix guidance."
keywords: ["REF-002", "broken markdown link", "references", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `REF-002`
- **Severity**: `HIGH`
- **Category**: `References`
- **Normative Level**: `SHOULD`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://spec.commonmark.org/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
[Spec](./missing-file.md)
```

### Valid

```markdown
[Spec](./VALIDATION-RULES.md)
```
