---
id: roo-002
title: "ROO-002: Invalid .roomodes Configuration"
sidebar_label: "ROO-002"
description: "agnix rule ROO-002 validates .roomodes configuration files. Severity: HIGH. See examples and fix guidance."
keywords: ["ROO-002", "invalid roomodes configuration", "roo code", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `ROO-002`
- **Severity**: `HIGH`
- **Category**: `Roo Code`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `roo-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.roocode.com/features/custom-modes

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "customModes": "not-an-array"
}
```

### Valid

```json
{
  "customModes": [
    {
      "slug": "designer",
      "name": "Designer",
      "roleDefinition": "You are a UI/UX designer.",
      "groups": ["read", "edit"]
    }
  ]
}
```
