---
id: roo-001
title: "ROO-001: Empty Roo Code Rule File - roo-code"
sidebar_label: "ROO-001"
description: "agnix rule ROO-001 checks for empty roo code rule file in roo-code files. Severity: HIGH. See examples and fix guidance."
keywords: ["ROO-001", "empty roo code rule file", "roo-code", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `ROO-001`
- **Severity**: `HIGH`
- **Category**: `roo-code`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `roo-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.roocode.com/features/custom-modes

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
Configuration omitted required fields for this rule.
```

### Valid

```text
# General Rules

Follow project coding standards and write clean, maintainable code.
```
