---
id: roo-004
title: "ROO-004: Invalid Mode Slug in Rule Directory - roo-code"
sidebar_label: "ROO-004"
description: "agnix rule ROO-004 checks for invalid mode slug in rule directory in roo-code files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["ROO-004", "invalid mode slug in rule directory", "roo-code", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `ROO-004`
- **Severity**: `MEDIUM`
- **Category**: `roo-code`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `roo-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.roocode.com/features/custom-modes

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
.roo/rules-INVALID SLUG/general.md
```

### Valid

```text
.roo/rules-architect/general.md
```
