---
id: roo-003
title: "ROO-003: Invalid .rooignore File - Roo Code"
sidebar_label: "ROO-003"
description: "agnix rule ROO-003 checks for invalid .rooignore file in roo code files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["ROO-003", "invalid .rooignore file", "roo code", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `ROO-003`
- **Severity**: `MEDIUM`
- **Category**: `Roo Code`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `roo-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.roocode.com/features/rooignore

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
Configuration omitted required fields for this rule.
```

### Valid

```text
node_modules/
*.log
.env
build/
```
