---
id: roo-005
title: "ROO-005: Invalid .roo/mcp.json Configuration - Roo Code"
sidebar_label: "ROO-005"
description: "agnix rule ROO-005 checks for invalid .roo/mcp.json configuration in roo code files. Severity: HIGH. See examples and fix guidance."
keywords: ["ROO-005", "invalid .roo/mcp.json configuration", "roo code", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `ROO-005`
- **Severity**: `HIGH`
- **Category**: `Roo Code`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `roo-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.roocode.com/features/mcp/using-mcp-in-roo

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "mcpServers": "not-an-object"
}
```

### Valid

```json
{
  "mcpServers": {
    "my-server": {
      "command": "node",
      "args": ["server.js"]
    }
  }
}
```
