---
id: roo-006
title: "ROO-006: Mode Slug Not Recognized - roo-code"
sidebar_label: "ROO-006"
description: "agnix rule ROO-006 checks for mode slug not recognized in roo-code files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["ROO-006", "mode slug not recognized", "roo-code", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `ROO-006`
- **Severity**: `MEDIUM`
- **Category**: `roo-code`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `roo-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.roocode.com/features/custom-modes

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
.roo/rules-custom-undefined/SKILL.md
```

### Valid

```text
.roo/rules-code/SKILL.md with content
```
