---
id: ver-001
title: "VER-001: No Tool/Spec Versions Pinned - Version Awareness"
sidebar_label: "VER-001"
description: "agnix rule VER-001 checks for no tool/spec versions pinned in version awareness files. Severity: LOW. See examples and fix guidance."
keywords: ["VER-001", "no tool/spec versions pinned", "version awareness", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `VER-001`
- **Severity**: `LOW`
- **Category**: `Version Awareness`
- **Normative Level**: `BEST_PRACTICE`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/avifenesh/agnix/issues/104

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
Pin MCP schema to an outdated version without fallback behavior.
```

### Valid

```markdown
Declare supported version range and degrade gracefully outside the range.
```
