---
id: ver-001
title: "VER-001: No Tool/Spec Versions Pinned - Version Awareness"
sidebar_label: "VER-001"
description: "agnix rule VER-001 checks for no tool/spec versions pinned in version awareness files. Severity: LOW. See examples and fix guidance."
keywords: ["VER-001", "no tool/spec versions pinned", "version awareness", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `VER-001`
- **Severity**: `LOW`
- **Category**: `Version Awareness`
- **Normative Level**: `BEST_PRACTICE`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/avifenesh/agnix/issues/104

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# .agnix.toml

[rules]
prompt_engineering = true

# No tool_versions or spec_revisions pinned```

### Valid

```markdown
# .agnix.toml

[tool_versions]
claude_code = "1.0.20"

[spec_revisions]
agent_skills_spec = "2025-01-15"```
