---
id: ws-004
title: "WS-004: Legacy .windsurfrules File Detected - Windsurf"
sidebar_label: "WS-004"
description: "agnix rule WS-004 checks for legacy .windsurfrules file detected in windsurf files. Severity: LOW. See examples and fix guidance."
keywords: ["WS-004", "legacy .windsurfrules file detected", "windsurf", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `WS-004`
- **Severity**: `LOW`
- **Category**: `Windsurf`
- **Normative Level**: `BEST_PRACTICE`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `windsurf`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.windsurf.com/windsurf/cascade/memories

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# .windsurfrules (legacy file)
Use strict mode.
```

### Valid

```markdown
# .windsurf/rules/typescript.md
# TypeScript Guidelines
Use strict mode.
```
