---
id: ws-002
title: "WS-002: Windsurf Rule File Exceeds Character Limit"
sidebar_label: "WS-002"
description: "agnix rule WS-002 checks for windsurf rule file exceeds character limit in windsurf files. Severity: HIGH. See examples and fix guidance."
keywords: ["WS-002", "windsurf rule file exceeds character limit", "windsurf", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `WS-002`
- **Severity**: `HIGH`
- **Category**: `windsurf`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `windsurf`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.windsurf.com/windsurf/cascade/memories

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```text
# Very long rule file exceeding 12000 characters...
```

### Valid

```text
# TypeScript Guidelines
Use strict mode and explicit types.
```
