---
id: ws-001
title: "WS-001: Empty Windsurf Rule File - Windsurf"
sidebar_label: "WS-001"
description: "agnix rule WS-001 checks for empty windsurf rule file in windsurf files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["WS-001", "empty windsurf rule file", "windsurf", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `WS-001`
- **Severity**: `MEDIUM`
- **Category**: `Windsurf`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `windsurf`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.windsurf.com/windsurf/cascade/memories

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown

```

### Valid

```markdown
# TypeScript Guidelines
Use strict mode and explicit types.
```
