---
id: ws-003
title: "WS-003: Empty or Oversized Windsurf Workflow File - Windsurf"
sidebar_label: "WS-003"
description: "agnix rule WS-003 checks for empty or oversized windsurf workflow file in windsurf files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["WS-003", "empty or oversized windsurf workflow file", "windsurf", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `WS-003`
- **Severity**: `MEDIUM`
- **Category**: `Windsurf`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-14`

## Applicability

- **Tool**: `windsurf`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.windsurf.com/windsurf/cascade/memories

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown

```

### Valid

```markdown
# Deploy Workflow
Run deployment steps.
```
