---
id: xml-003
title: "XML-003: Unmatched Closing Tag - XML"
sidebar_label: "XML-003"
description: "agnix rule XML-003 checks for unmatched closing tag in xml files. Severity: HIGH. See examples and fix guidance."
keywords: ["XML-003", "unmatched closing tag", "xml", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XML-003`
- **Severity**: `HIGH`
- **Category**: `XML`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://www.w3.org/TR/xml/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```xml
<analysis><rule id="XML-001"></analysis>
```

### Valid

```xml
<analysis><rule id="XML-001">ok</rule></analysis>
```
