---
id: xml-002
title: "XML-002: Mismatched Closing Tag - XML"
sidebar_label: "XML-002"
description: "agnix rule XML-002 checks for mismatched closing tag in xml files. Severity: HIGH. See examples and fix guidance."
keywords: ["XML-002", "mismatched closing tag", "xml", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XML-002`
- **Severity**: `HIGH`
- **Category**: `XML`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://www.w3.org/TR/xml/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```xml
<outer>
<inner>
Content here.
</outer>
</inner>```

### Valid

```xml
<outer>
<inner>
Content here.
</inner>
</outer>```
