---
id: xp-007
title: "XP-007: AGENTS.md Exceeds Codex Byte Limit - Cross-Platform"
sidebar_label: "XP-007"
description: "agnix rule XP-007 checks for agents.md exceeds codex byte limit in cross-platform files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["XP-007", "agents.md exceeds codex byte limit", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-007`
- **Severity**: `MEDIUM`
- **Category**: `Cross-Platform`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2025-05-01`

## Applicability

- **Tool**: `codex-cli`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# AGENTS.md

(content exceeding 32768 bytes - will be truncated by Codex CLI)
```

### Valid

```markdown
# AGENTS.md

## Overview
Concise project instructions under 32KB.

## Commands
- npm run build
- npm run test
```
