---
id: xp-004
title: "XP-004: Conflicting Build/Test Commands - Cross-Platform"
sidebar_label: "XP-004"
description: "agnix rule XP-004 checks for conflicting build/test commands in cross-platform files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["XP-004", "conflicting build/test commands", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-004`
- **Severity**: `MEDIUM`
- **Category**: `Cross-Platform`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# CLAUDE.md

## Commands
- `npm install`
- `npm run build`
- `npm run test`

# AGENTS.md (separate file, same project)

## Commands
- `pnpm install`
- `pnpm build`
- `pnpm test````

### Valid

```markdown
# CLAUDE.md

## Commands
- `npm install`
- `npm run build`
- `npm test`

# AGENTS.md (separate file, same project)

## Commands
- `npm install`
- `npm run build`
- `npm test````
