---
id: xp-002
title: "XP-002: AGENTS.md Platform Compatibility - Cross-Platform"
sidebar_label: "XP-002"
description: "agnix rule XP-002 checks for agents.md platform compatibility in cross-platform files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["XP-002", "agents.md platform compatibility", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-002`
- **Severity**: `MEDIUM`
- **Category**: `Cross-Platform`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-09`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://opencode.ai/docs/rules
- https://cursor.com/docs/context/rules
- https://docs.cline.bot/prompting/cline-memory-bank#cline-memory-bank-custom-instructions-[copy-this]
- https://github.blog/changelog/2025-08-28-copilot-coding-agent-now-supports-agents-md-custom-instructions/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
Just plain text without any markdown headers or structure.
```

### Valid

```markdown
# Project Memory

## Build Commands

### Testing

Run tests with npm test.
```
