---
id: xp-005
title: "XP-005: Conflicting Tool Constraints - Cross-Platform"
sidebar_label: "XP-005"
description: "agnix rule XP-005 checks for conflicting tool constraints in cross-platform files. Severity: HIGH. See examples and fix guidance."
keywords: ["XP-005", "conflicting tool constraints", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-005`
- **Severity**: `HIGH`
- **Category**: `Cross-Platform`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# CLAUDE.md

allowed-tools: Read Write Edit Bash Grep
You can use Bash for git operations.

# AGENTS.md (separate file, same project)

Never use Bash for any operations.
Do not use Edit or Write tools.```

### Valid

```markdown
# CLAUDE.md

allowed-tools: Read Write Edit Bash Grep

# AGENTS.md (separate file, same project)

allowed-tools: Read Write Edit Bash Grep```
