---
id: xp-003
title: "XP-003: Hard-Coded Platform Paths - Cross-Platform"
sidebar_label: "XP-003"
description: "agnix rule XP-003 checks for hard-coded platform paths in cross-platform files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["XP-003", "hard-coded platform paths", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-003`
- **Severity**: `MEDIUM`
- **Category**: `Cross-Platform`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Configuration

- Claude: .claude/settings.json
- Cursor: .cursor/rules/
- Config: ~/.config/my-tool/config.yaml
```

### Valid

```markdown
# Configuration

Use environment variables for all platform-specific settings.
See ./src/config.ts for defaults.
```
