---
id: xp-001
title: "XP-001: Platform-Specific Feature in Generic Config"
sidebar_label: "XP-001"
description: "agnix rule XP-001 checks for platform-specific feature in generic config in cross-platform files. Severity: HIGH. See examples and fix guidance."
keywords: ["XP-001", "platform-specific feature in generic config", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-001`
- **Severity**: `HIGH`
- **Category**: `Cross-Platform`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Project Guidelines

---
context: fork
agent: reviewer
allowed-tools: Read Write
---

- type: PreToolExecution
  command: echo "lint"
```

### Valid

```markdown
# Project Guidelines

Follow the coding style guide.

## Commands
- npm run build
- npm run test
```
