---
id: xp-006
title: "XP-006: Multiple Layers Without Documented Precedence"
sidebar_label: "XP-006"
description: "agnix rule XP-006 checks for multiple layers without documented precedence in cross-platform files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["XP-006", "multiple layers without documented precedence", "cross-platform", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XP-006`
- **Severity**: `MEDIUM`
- **Category**: `Cross-Platform`
- **Normative Level**: `SHOULD`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
Use only CLAUDE.md instructions and ignore AGENTS.md.
```

### Valid

```markdown
Use both CLAUDE.md and AGENTS.md with explicit precedence and conflict handling.
```
