---
id: as-016
title: "AS-016: Skill Parse Error - Agent Skills"
sidebar_label: "AS-016"
description: "agnix rule AS-016 checks for skill parse error in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-016", "skill parse error", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-016`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
---
description: Deploys production changes
---

# deploy
Use the skill now.
```

### Valid

```markdown
---
name: deploy-prod
description: Deploy production with explicit checks
---

# deploy-prod
Run rollout checks before deployment.
```
