---
id: cdx-001
title: "CDX-001: Invalid Approval Mode - Codex CLI"
sidebar_label: "CDX-001"
description: "agnix rule CDX-001 checks for invalid approval mode in codex cli files. Severity: HIGH. See examples and fix guidance."
keywords: ["CDX-001", "invalid approval mode", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-001`
- **Severity**: `HIGH`
- **Category**: `Codex CLI`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/openai/codex

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```text
Configuration omitted required fields for this rule.```

### Valid

```text
Configuration includes required fields and follows the rule.```
