---
id: cop-003
title: "COP-003: Invalid Glob Pattern in applyTo - GitHub Copilot"
sidebar_label: "COP-003"
description: "agnix rule COP-003 checks for invalid glob pattern in applyto in github copilot files. Severity: HIGH. See examples and fix guidance."
keywords: ["COP-003", "invalid glob pattern in applyto", "github copilot", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `COP-003`
- **Severity**: `HIGH`
- **Category**: `GitHub Copilot`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `github-copilot`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.github.com/en/copilot/customizing-copilot

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Copilot Instructions
Write whatever code seems fine.
```

### Valid

```markdown
# Copilot Instructions
Use project coding standards and keep tests updated.
```
