---
id: cur-003
title: "CUR-003: Invalid YAML Frontmatter - Cursor"
sidebar_label: "CUR-003"
description: "agnix rule CUR-003 checks for invalid yaml frontmatter in cursor files. Severity: HIGH. See examples and fix guidance."
keywords: ["CUR-003", "invalid yaml frontmatter", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-003`
- **Severity**: `HIGH`
- **Category**: `Cursor`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Rule
No metadata block
```

### Valid

```markdown
---
description: Cursor rule for repository policy
---
Use project-specific guidance.
```
