---
id: cur-007
title: "CUR-007: alwaysApply with Redundant globs - Cursor"
sidebar_label: "CUR-007"
description: "agnix rule CUR-007 checks for alwaysapply with redundant globs in cursor files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CUR-007", "alwaysapply with redundant globs", "cursor", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CUR-007`
- **Severity**: `MEDIUM`
- **Category**: `Cursor`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `Yes (safe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `cursor`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Rule
No metadata block
```

### Valid

```markdown
---
description: Cursor rule for repository policy
---
Use project-specific guidance.
```
