---
id: as-004
title: "AS-004: Invalid Name Format - Agent Skills"
sidebar_label: "AS-004"
description: "agnix rule AS-004 checks for invalid name format in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-004", "invalid name format", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-004`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (safe/unsafe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `1.0`

## Evidence Sources

- https://agentskills.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: Run_Tests!
description: Use when running the test suite
---
Execute all unit tests.
```

### Valid

```markdown
---
name: run-tests
description: Use when running the test suite
---
Execute all unit tests.
```
