---
id: cdx-003
title: "CDX-003: AGENTS.override.md in Version Control - Codex CLI"
sidebar_label: "CDX-003"
description: "agnix rule CDX-003 checks for agents.override.md in version control in codex cli files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["CDX-003", "agents.override.md in version control", "codex cli", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `CDX-003`
- **Severity**: `MEDIUM`
- **Category**: `Codex CLI`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `codex`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://github.com/openai/codex

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```toml
# AGENTS.override.md

Local overrides that should not be committed to version control.
```

### Valid

```toml
# AGENTS.md

Project instructions for Codex CLI.
```
