---
id: pe-001
title: "PE-001: Lost in the Middle - Prompt Engineering"
sidebar_label: "PE-001"
description: "agnix rule PE-001 checks for lost in the middle in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-001", "lost in the middle", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-001`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://aclanthology.org/2024.tacl-1.9/

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
## Guidelines

Line 1
Line 2
Line 3
Line 4
Line 5
Line 6
Line 7
Line 8
NEVER commit secrets to the repository.
Line 10
Line 11
Line 12
Line 13
Line 14
Line 15
Line 16
Line 17
Line 18
Line 19
Line 20
```

### Valid

```markdown
# Critical Rules

NEVER commit secrets to the repository.
Always run tests before pushing.

## Guidelines

Line 1
Line 2
Line 3
Line 4
Line 5
Line 6
Line 7
Line 8
Line 9
Line 10
Line 11
Line 12
Line 13
Line 14
Line 15
Line 16
Line 17
Line 18
```
