---
id: pe-004
title: "PE-004: Ambiguous Instructions - Prompt Engineering"
sidebar_label: "PE-004"
description: "agnix rule PE-004 checks for ambiguous instructions in prompt engineering files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["PE-004", "ambiguous instructions", "prompt engineering", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `PE-004`
- **Severity**: `MEDIUM`
- **Category**: `Prompt Engineering`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/overview

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Rules

Usually format output as JSON.
Include tests if possible.
Sometimes run the linter when appropriate.
```

### Valid

```markdown
# Rules

Always format output as JSON.
Never skip test execution.
```
