---
id: agm-005
title: "AGM-005: Platform-Specific Features Without Guard"
sidebar_label: "AGM-005"
description: "agnix rule AGM-005 checks for platform-specific features without guard in agents.md files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AGM-005", "platform-specific features without guard", "agents.md", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AGM-005`
- **Severity**: `MEDIUM`
- **Category**: `AGENTS.md`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://developers.openai.com/codex/guides/agents-md
- https://docs.cursor.com/en/context

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Instructions
Do everything automatically.
```

### Valid

```markdown
## Project Instructions
- Use AGENTS.md as instruction entrypoint
- Keep commands explicit and test changes
```
