---
id: agm-003
title: "AGM-003: Character Limit (Windsurf) - AGENTS.md"
sidebar_label: "AGM-003"
description: "agnix rule AGM-003 checks for character limit (windsurf) in agents.md files. Severity: MEDIUM. See examples and fix guidance."
keywords: ["AGM-003", "character limit (windsurf)", "agents.md", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AGM-003`
- **Severity**: `MEDIUM`
- **Category**: `AGENTS.md`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `windsurf`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://docs.windsurf.com/windsurf/cascade/memories

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```markdown
# Instructions
Do everything automatically.
```

### Valid

```markdown
## Project Instructions
- Use AGENTS.md as instruction entrypoint
- Keep commands explicit and test changes
```
