---
id: mcp-011
title: "MCP-011: Invalid MCP server type - MCP"
sidebar_label: "MCP-011"
description: "agnix rule MCP-011 checks for invalid mcp server type in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-011", "invalid mcp server type", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-011`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-07`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-03-26`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```json
{
  "jsonrpc": "1.0",
  "tools": []
}
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "tools": [
    {
      "name": "validate_file",
      "description": "Validate one configuration file",
      "inputSchema": {"type": "object"}
    }
  ]
}
```
