---
id: mcp-007
title: "MCP-007: MCP Parse Error - MCP"
sidebar_label: "MCP-007"
description: "agnix rule MCP-007 checks for mcp parse error in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-007", "mcp parse error", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-007`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-06-18`

## Evidence Sources

- https://modelcontextprotocol.io/specification

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples are illustrative snippets for this rule category.

### Invalid

```json
{
  "jsonrpc": "1.0",
  "tools": []
}
```

### Valid

```json
{
  "jsonrpc": "2.0",
  "tools": [
    {
      "name": "validate_file",
      "description": "Validate one configuration file",
      "inputSchema": {"type": "object"}
    }
  ]
}
```
