---
id: mcp-004
title: "MCP-004: Missing Tool Description - MCP"
sidebar_label: "MCP-004"
description: "agnix rule MCP-004 checks for missing tool description in mcp files. Severity: HIGH. See examples and fix guidance."
keywords: ["MCP-004", "missing tool description", "mcp", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `MCP-004`
- **Severity**: `HIGH`
- **Category**: `MCP`
- **Normative Level**: `SHOULD`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `2025-06-18`

## Evidence Sources

- https://modelcontextprotocol.io/docs/concepts/tools

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```json
{
  "name": "deploy",
  "description": "Short",
  "inputSchema": { "type": "object" }
}
```

### Valid

```json
{
  "name": "deploy",
  "description": "Deploys the application to the specified environment with rollback support",
  "inputSchema": { "type": "object" }
}
```
