---
id: as-007
title: "AS-007: Reserved Name - Agent Skills"
sidebar_label: "AS-007"
description: "agnix rule AS-007 checks for reserved name in agent skills files. Severity: HIGH. See examples and fix guidance."
keywords: ["AS-007", "reserved name", "agent skills", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `AS-007`
- **Severity**: `HIGH`
- **Category**: `Agent Skills`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `claude-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://platform.claude.com/docs

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
---
name: claude
description: Use when the user needs help with tasks
---
Provide assistance.
```

### Valid

```markdown
---
name: my-helper
description: Use when the user needs help with tasks
---
Provide assistance.
```
