---
id: ref-001
title: "REF-001: Import File Not Found - References"
sidebar_label: "REF-001"
description: "agnix rule REF-001 checks for import file not found in references files. Severity: HIGH. See examples and fix guidance."
keywords: ["REF-001", "import file not found", "references", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `REF-001`
- **Severity**: `HIGH`
- **Category**: `References`
- **Normative Level**: `MUST`
- **Auto-Fix**: `No`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `claude-code`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://code.claude.com/docs/en/memory

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```markdown
# Project

See @docs/nonexistent-file.md for the style guide.

(where docs/nonexistent-file.md does not exist)
```

### Valid

```markdown
# Project

See @docs/coding-standards.md for style guide.

(where docs/coding-standards.md exists on disk)
```
