---
id: xml-001
title: "XML-001: Unclosed XML Tag - XML"
sidebar_label: "XML-001"
description: "agnix rule XML-001 checks for unclosed xml tag in xml files. Severity: HIGH. See examples and fix guidance."
keywords: ["XML-001", "unclosed xml tag", "xml", "validation", "agnix", "linter"]
---

## Summary

- **Rule ID**: `XML-001`
- **Severity**: `HIGH`
- **Category**: `XML`
- **Normative Level**: `MUST`
- **Auto-Fix**: `Yes (unsafe)`
- **Verified On**: `2026-02-04`

## Applicability

- **Tool**: `all`
- **Version Range**: `unspecified`
- **Spec Revision**: `unspecified`

## Evidence Sources

- https://platform.claude.com/docs

## Test Coverage Metadata

- Unit tests: `true`
- Fixture tests: `true`
- E2E tests: `false`

## Examples

The following examples demonstrate what triggers this rule and how to fix it.

### Invalid

```xml
<example>
This XML tag is never closed and will confuse the parser.
```

### Valid

```xml
<example>
This is a properly closed XML tag in markdown.
</example>
```
